packages/server/lib/browsers/geckodriver/index.ts

```diff
@@ -0,0 +1,119 @@
+import Bluebird from 'bluebird'
+import debugModule from 'debug'
+import errors from '@packages/errors'
+import type { ChildProcess } from 'child_process'
+
+const geckoDriverPackageName = 'geckodriver'
+const GECKODRIVER_DEBUG_NAMESPACE = 'cypress:server:browsers:geckodriver'
+const GECKODRIVER_DEBUG_NAMESPACE_VERBOSE = 'cypress-verbose:server:browsers:geckodriver'
+const debug = debugModule(GECKODRIVER_DEBUG_NAMESPACE)
+const debugVerbose = debugModule(GECKODRIVER_DEBUG_NAMESPACE_VERBOSE)
+
+type SpawnOpt = { [key: string]: string | string[] | SpawnOpt }
+
+export type StartGeckoDriverArgs = {
+  host: string
+  port: number
+  marionetteHost: string
+  marionettePort: number
+  webdriverBidiPort: number
+  profilePath: string
+  binaryPath: string
+  spawnOpts?: SpawnOpt
+}
+
+/**
+ * Class with static methods of the GeckoDriver as it returns a ChildProcess
```

Comment:
I think this comment could be re-worded.

Reply:
addressed in fb54eaaa9a